src/Microsoft.VisualStudio.Extensibility.Testing.Xunit.Shared/Harness/IdeTestAssemblyRunner.cs

```diff
@@ -455,7 +456,12 @@ public bool OnMessage(IMessageSinkMessage message)
                             testCaseFinished.TestsSkipped + testCaseFinished.TestsFailed);
                     }
 
-                    return !_cancellationToken.IsCancellationRequested;
+                    if (_cancellationToken.IsCancellationRequested)
+                    {
+                        return false;
+                    }
+
+                    return _messageSink.OnMessage(message);
```

Comment:
this is the fix.  the newish xunit runner is expecting that if we call `_messageSink.OnMessage(message)` for an `ITestCaseStarting` message (adds to a cache), we also call it for `ITestCaseFinished` (removes from cache).

Otherwise if we don't call it for `ITestCaseFinished` we get an exception when calling it for `ITestCaseStarting` on the retry.